ql/src/java/org/apache/hadoop/hive/ql/plan/DynamicPartitionCtx.java

```diff
@@ -50,6 +52,14 @@
   private String defaultPartName; // default partition name in case of null or empty value
   private int maxPartsPerNode;    // maximum dynamic partitions created per mapper/reducer
   private Pattern whiteListPattern;
+  /**
+   * Expressions describing a custom way of sorting the table before write. Expressions can reference simple
+   * column descriptions or a tree of expressions containing more columns and UDFs.
+   * Can be useful for custom bucket/hash sorting.
+   * A custom expression should be a lambda that is given the original column description expressions as per read
+   * schema and returns a single expression. Example for simply just referencing column 3: cols -> cols.get(3).clone()
+   */
+  private transient List<Function<List<ExprNodeDesc>, ExprNodeDesc>> customSortExpressions;
```

Comment:
Out of curiosity, why transient?

Reply:
The job has the dpCtx serialized, and for some reason kryo is throwing exceptions for lambdas on the executor side.
The expressions here are only needed in HS2 during query planning so I just prevent them from going to executors in the first place.

Reply:
Yeah, I was wondering if it's okay if these expressions don't get to the executors. But I got my answer - Thanks!